integration_tests/e2e/overview/goal/completeGoal.cy.ts

```diff
@@ -0,0 +1,152 @@
+import { urlEqualTo } from '../../../mockApis/wiremock/matchers/url'
+import { matchingJsonPath } from '../../../mockApis/wiremock/matchers/content'
+import GoalsPage from '../../../pages/goal/GoalsPage'
+import CompleteOrArchiveGoalPage from '../../../pages/goal/CompleteOrArchiveGoalPage'
+import CompleteOrArchiveGoalValue from '../../../../server/enums/CompleteOrArchiveGoalValue'
+import CompleteGoalPage from '../../../pages/goal/CompleteGoalPage'
+
+context('Complete a goal', () => {
+  const prisonNumber = 'G6115VJ'
+  const goalReference = '10efc562-be8f-4675-9283-9ede0c19dade'
+
+  beforeEach(() => {
+    cy.task('reset')
+    cy.task('stubSignInAsUserWithEditAuthority')
+    cy.task('stubAuthUser')
+    cy.task('stubGetHeaderComponent')
+    cy.task('stubGetFooterComponent')
+    cy.task('stubPrisonerList')
+    cy.task('stubCiagInductionList')
+    cy.task('stubActionPlansList')
+    cy.task('getPrisonerById')
+    cy.task('stubGetInduction')
+    cy.task('getActionPlan')
+    cy.task('stubLearnerProfile')
+    cy.task('stubLearnerEducation')
+    cy.task('archiveGoal')
+    cy.task('stubGetAllPrisons')
+  })
+
+  it('should be able to navigate directly to complete goal page', () => {
+    // Given
+    cy.signIn()
+
+    // When
+    cy.visit(`/plan/${prisonNumber}/goals/${goalReference}/complete`)
+
+    // Then
+    const completeGoalPage = Page.verifyOnPage(CompleteGoalPage)
+    completeGoalPage.isForGoal(goalReference)
+  })
+
+  it('should be able to navigate to complete goals page from overview', () => {
+    // Given
+    cy.signIn()
+
+    cy.visit(`/plan/${prisonNumber}/view/overview`)
+    const overviewPage = Page.verifyOnPage(OverviewPage)
+
+    // When
+    const goalsPage = overviewPage.clickViewInProgressGoalsButton()
+    const completeOrArchivePage = goalsPage //
+      .checkOnInProgressGoalsTab()
+      .clickCompleteOrArchiveButtonForGoal(goalReference)
+    Page.verifyOnPage(CompleteOrArchiveGoalPage)
+    completeOrArchivePage.selectOption(CompleteOrArchiveGoalValue.COMPLETE).submitPage()
+
+    // Then
+    Page.verifyOnPage(CompleteGoalPage)
+  })
+
+  it('should ask for confirmation and return to overview if choosing no', () => {
+    // Given
+    cy.signIn()
+    cy.visit(`/plan/${prisonNumber}/goals/${goalReference}/complete`)
+    const completeGoalPage = Page.verifyOnPage(CompleteGoalPage)
+    completeGoalPage //
+      .clickNo()
+    // Then
+    Page.verifyOnPage(OverviewPage)
+    cy.wiremockVerifyNoInteractions(
+      putRequestedFor(urlEqualTo(`/action-plans/${prisonNumber}/goals/${goalReference}/complete`)),
+    )
+  })
+
+  it('Should be able to complete a goal successfully without entering the optional notes', () => {
+    // Given
+    cy.task('completeGoal', { prisonNumber, goalReference })
+    cy.signIn()
+    cy.visit(`/plan/${prisonNumber}/view/goals`)
+    const goalsPage = Page.verifyOnPage(GoalsPage)
+
+    // When
+    const completeOrArchiveGoalPage = goalsPage //
```

Comment:
I think our `Goals` page driver class has got a bit big / does too much / concerns itself with 3 tabs that are showing fundamentally different goals.
I found lines 87 and 88, and 122 and 123 difficult to read and reason about, because I was expecting the state of the `Goals` page to be on the In-Progress tab showing in-progress goals. But we are apparently checking the archived goals are displayed and there are 2 of them!

I know why it works (because the `Goals` page class knows about all 3 tabs, irrespective of whether they are currently rendered/visible or not), but reading the test is hard

I'd be tempted to remove lines 87 and 88, and 122 and 123 as I dont think they are pertinent to competing a goal?

And I think my PR (and one I will do shortly) will make this easier in the future (I'll do a PR that splits out the `Goals` page drive class into 3, one for each tab. Then it will be much clearer which tab we are on and what functionality we can expect 👍 )